pkg/operator/bootstrapteardown/bootstrap_teardown_controller.go

```diff
@@ -0,0 +1,306 @@
+			// if any 1 kube-apiserver pod has more than 1
+			klog.V(4).Info("kube-apiserver has required values")
+			return true
+		}
+	}
+	return false
+}
+
+type ConfigData struct {
+	StorageConfig struct {
+		Urls []string
+	}
+}
+
+func (c *BootstrapTeardownController) configMapHasRequiredValues(configMap *corev1.ConfigMap) bool {
+	config, ok := configMap.Data[configMapKey]
+	if !ok {
+		klog.V(2).Infof("configMapHasRequiredValues: config.yaml key missing configmap %s/%s", configMap.Namespace, configMap.Name)
+		c.eventRecorder.Eventf("KASconfigmapDoesNotHaveRequiredValues", "configMapHasRequiredValues: config.yaml key missing for configmap %s/%s", configMap.Namespace, configMap.Name)
+		return false
+	}
+	var configData ConfigData
+	err := json.Unmarshal([]byte(config), &configData)
+	if err != nil {
+		c.eventRecorder.Eventf("KASconfigmapDoesNotHaveRequiredValues", "error unmarshalling configmap %s/%s data: %#v", configMap.Namespace, configMap.Name, err)
+		return false
+	}
+	if len(configData.StorageConfig.Urls) == 0 {
+		c.eventRecorder.Eventf("KASconfigmapDoesNotHaveRequiredValues", "configMapHasRequiredValues: length of storageUrls %#v is 0 for configmap %s/%s", configData.StorageConfig.Urls, configMap.Namespace, configMap.Name)
+		return false
+	}
+	if len(configData.StorageConfig.Urls) == 1 &&
+		!strings.Contains(configData.StorageConfig.Urls[0], "etcd") {
+		c.eventRecorder.Eventf("KASconfigmapDoesNotHaveRequiredValues", "configMapHasRequiredValues: config %s/%s has a single IP: %#v", configMap.Namespace, configMap.Name, strings.Join(configData.StorageConfig.Urls, ", "))
+		return false
+	}
+	c.eventRecorder.Eventf("KASconfigmapDoesHasRequiredValues", "configMapHasRequiredValues: return true for %s/%s", configMap.Namespace, configMap.Name)
+	return true
+}
+
+func (c *BootstrapTeardownController) Run(stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	klog.Infof("Starting BootstrapTeardownController")
+	defer klog.Infof("Shutting down BootstrapTeardownController")
+
+	if !cache.WaitForCacheSync(stopCh, c.cachesToSync...) {
+		return
+	}
+
+	go wait.Until(c.runWorker, time.Second, stopCh)
+
+	// add time based trigger
+	go wait.PollImmediateUntil(time.Minute, func() (bool, error) {
+		c.queue.Add(workQueueKey)
+		return false, nil
+	}, stopCh)
+
+	<-stopCh
+}
+
+// this function checks if etcd has scaled the initial bootstrap cluster
+// with all 3 masters
+func (c *BootstrapTeardownController) isEtcdAvailable(currentEtcdOperator *operatorv1.Etcd) (bool, error) {
+	if currentEtcdOperator.Spec.ManagementState == operatorv1.Unmanaged {
+		c.eventRecorder.Eventf("OperatorUnmanaged", "cluster etcd operator is in Unmanaged mode")
+		return true, nil
+	}
+	if operatorv1helpers.IsOperatorConditionTrue(currentEtcdOperator.Status.Conditions, clustermembercontroller.ConditionBootstrapSafeToRemove) {
+		c.eventRecorder.Eventf("EtcdBootstrapped", "cluster etcd operator bootstrapped successfully")
```

Comment:
this is steady state. v(2) after the first transition.  You may track that transition in the controller so we get one per start if you wish.